### PR TITLE
Update list search timestamp to only update if resource not deleted

### PR DESCRIPTION
**Pull request by Sigmonia**
#### Rationale
[Issue #48878](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=48878) Search indexer has race condition

#### Changes
* Update list indexer to only update if item resource is present

### CODE REVIEW

File: list/src/org/labkey/list/model/ListManager.java [hunk 1]
```diff
@@ -1088,8 +1088,13 @@ private void setItemLastIndexed(ListDefinition list, Object pk, TableInfo ti, lo
             if (null != keyColumn)
             {
                 String keySelectName = keyColumn.getSelectName();
-                new SqlExecutor(sti.getSchema()).execute("UPDATE " + getListTableName(sti) + " SET LastIndexed = ? WHERE " +
-                        keySelectName + " = ?", new Timestamp(ms), pk);
+
```
Comment (labkey-matthewb) [inline list/src/org/labkey/list/model/ListManager.java]: I think this additional condition in the WHERE clause is redundant.  The original already checks that the row exists.  Wasn't the bug that the actual list got deleted?

**Review by labkey-matthewb — CHANGES_REQUESTED** [review]

File: list/src/org/labkey/list/model/ListManager.java (hunk 1 above)
Comment (labkey-adam) [reply list/src/org/labkey/list/model/ListManager.java]: Correct... I discussed this with Ian. My suggestion is to catch, swallow constraint violations, and re-throw anything else.

**Review by labkey-adam — APPROVED** [review]

File: list/src/org/labkey/list/model/ListManager.java [hunk 2]
```diff
@@ -695,7 +695,18 @@ private void indexItems(@NotNull final IndexTask task, final ListDefinition list
...
                     {
-                        ListManager.get().setItemLastIndexed(list, pk, listTable, ms, modified);
+                        try
+                        {
+                            ListManager.get().setItemLastIndexed(list, pk, listTable, ms, modified);
+                        }
+                        catch (RuntimeSQLException e)
+                        {
+                            // This may occur due to a race condition between enumeration and list deletion. Issue #48878
+                            if (e.isConstraintException())
```
Comment (labkey-matthewb) [inline list/src/org/labkey/list/model/ListManager.java]: Has anyone tested that table not found throws an exception that is flagged as a constraint exception???  I might be, but I wouldn't expect that.

Comment (Sigmonia) [reply list/src/org/labkey/list/model/ListManager.java]: tested newest updated against both MS-SQL and psql.

**Review by labkey-matthewb — APPROVED** [review]

**Pull request merged**